### ISSUE TITLE
Suggestions for improvements and feedback

### ISSUE THREAD

**Issue by kaktux** [nikolaystrikhar/gutenberg-forms#124]
hi there,

as i am transfering an old wp page with some outdated plugins i tried out your form block as alternative.
An so far i love it - altough there is some place for improvement - so i wanted to give a little feedback.

First thing:
the link to the docs didn't work.

Which would help with some basic questions.
eg. what entries is for (could only drag stuff there) etc. - but guess this is about "more to come".
Also i couldnt really find out what difference it makes to use text or name fields.
What i am also currently trying to figure out is how to set up the email notification - order of data etc. as this doesn't work so far. But i just finished my first form - so this will be figured out with some try + error.

Second:
Feedback:
It would be nice to add other blocks to the gutenberg forms block that give some more options - like colours, button, colums/grids etc. A lot already exist with tons of nice options and maybe it this can be done without having to use those extra blocks from gutenberg forms. But i am no programmer. The column and others just seemed redundant to me. Thsoe offer some options i like - like space between colums, background colour, button settings like width, image etc. would have been nice. 

Thirdly:
 i wanted to create a contact form for inquiries to a company webpage. As this is an european company a consent to save the send info (name, phone, adress etc.) is needed.

So - although this can be done by hand - a template would be nice to have - as at least im thinking that i wont be the only one with this problem.

You could also alter an existing contact form. 

Mine looks like:

Headline
Company - Title
Lastname - Firstname
Adress
ZipCode - Region
Country
Phone
email
Website
Subject
Message
checkbox -> I consent to having NAME collect the data entered in this form.

Paragraph: For further information how we manage your data have a look at our privacy policy.
Button

So basically the consent checkbox -> marked as required 

Lastly:
The name: i only found this block by accident - as you find a lot of other stuff when you search for gutenberg form. So although it makes sense - it probably doesnt help in terms of letting people find this plugin. 

Thats all for now - thx for this amazing plugin. Which needs to be recognized much more - as i don't see much it doesn't offer compared to other plugins mentioned in "contact form plugins overview " articles.

**Comment by munirkamal** [comment]
Hey @kaktux 

Thank you so much for trying the plugin and taking the time to leave this feedback. Of course, there is a lot of room for improvement, this is yet a new developing plugin, to be honest. The features you currently see were mostly suggested by the users and fellows like you who have been kind to test and suggest features. Let me try to answer your questions.

1) Documentation: Well, this is something I need to work, I am just not getting much time for this. But, I will do it soon. Let me answer your queries here.

**Entries**
That area will save your form submission data simply. You can, however, disable it on any form if you do not wish to save the entries to the database. 

**Email Notification**
By default when you create a new form you already get all the data of the form fields to your email (unless you disable email notification). By default, you will see the "all data" tag, but you can remove it and create your own data that you like to receive in email.
![Screen Recording 2020-05-05 at 04 32 PM](https://user-images.githubusercontent.com/5817882/81062428-5c97c680-8eef-11ea-9d0d-6cb374ebf7b6.gif)

2) Insert Other Gutenberg Blocks in Form
This is already possible, you can add any Gutenberg blocks between form fields/blocks. That is the beauty of this form builder. However, for columns and grouping fields we have adding our own Form Columns & Form Group Blocks for a reason. 
![Screen Recording 2020-05-05 at 05 12 PM](https://user-images.githubusercontent.com/5817882/81064809-cc0fb500-8ef3-11ea-815c-3454be011201.gif)

3) Form Templates & Consent Field
Well, we have already started creating Form Templates and you may see the templates we have created so far. But will add more templates later. I am also thinking about an option for users to submit their templates to the library. **More on this later**.

Regarding the consent field, basically you can simply use the checkbox field block for this purpose and mark it required. There is no need for a separate field for this (unless I am missing something?). 

4) Name:
Isn't this is an accurate name for what people will search for? 
"**Gutenberg form**" 
 
What did you search for finding a form for Gutenberg? Or what name do you suggest? (although it won't change but interested to hear it out).

----------

Thank you again for testing the plugin and please keep sharing your suggestions and feedback to improve this plugin.

Regards,
Munir

**Comment by kaktux** [comment]
Thx a lot - now you alread wrote half of the docs ;). The link from the plugins (http://docs.gutenbergforms.com/) just doesnt work. In case you didnt notice.

About 3. thats how i solved it: with a (required) checkbox - it was just a suggestion because probably some will have to do the same - ask for consent to save your data due to european law. Same goes for the entries (which make sense).  But definitly is solvable already.

About 4 the name:
I dont think it is a bad name per se - it says what it is. What i am saying is that it is had to find with that name.
as i searched "gutenberg forms" there where several articles about how to create forms, plugins (lists that did not include this plugin) on top. Somewhere at the end of the first page was the github repository. But as i just tried this seems to have changed.

If i remember right i didnt search for gutenberg forms - but wordpress forms. As you can find mostly comparisons with that (or pages of other form plugins). But as i just added some comments under some of the newer articles it now can be found that way. Not because i am not affiliated nor a superfan as you might know - but this may help others like me to find and try out this nice plugin. 

so thx for all your work.

regards

**Comment by munirkamal** [comment]
Hey @kaktux 

Yes, the docs are not created yet. I will create a coming soon page. 

For the consent field. What exactly it does extra than a checkbox? Can you explain further? If there is a need I can create a separate field block for this.

Regarding the name, I see you actually concerned about SEO. This is obviously a new shiny plugin and I have done no efforts yet to get it covered by any blogs out there. The main focus right now is to make it a solid feature-rich plugin that users find very useful compared to other legacy form builders. 

Obviously in those "**Best WordPress Forms**" kind of review posts, you will only see the top paid plugins that offers good affiliate commissions or sponsor the post. 😊

We will get some real coverage (like you did) hopefully by the happy users who can see the potential in this plugin. 😇

Cheers,
Munir

**Comment by macemmek** [comment]
@munirkamal The concept of your plugin is so good that I hope you'll be paid for that like in freemium model or so. Keep up the good work!

**Comment by munirkamal** [comment]
@macemmek 

Thank you mate. I hope so too. 🤑

**Comment by kaktux** [comment]
"For the consent field. What exactly it does extra than a checkbox? Can you explain further? If there is a need I can create a separate field block for this."
nothing its basically a checkbox. No extra field needed. Just a required checkbox that needs to be checked before form can be submitted.

I finally got mine working. I tried to use an extra button to submit bofore (didnt work) - but with the form-block-button enabled the form works.

So another quick feedback:
1. Would be nice if the normal button could be styled separatly. By default it seems to get colour from the theme. But if you change the colour in form it takes those also - mainly the background goes white if you choose so. Also
2. Field Background Color -> only works in Email, Subject, Message (not in Name, Text, Phone, Website)
3. Email, Subject and Message field are smaller (width) then all others.
4. Form builder starts with form-Block already inserted. Didnt find a way to start with a container/grid etc.. But maybe you mean it like that: builder to create shortcode - or else insert into page directly.
5. i tried to add a form into a sidebar (using Off-Canvas Sidebars & Menus (Slidebars) (https://wordpress.org/plugins/off-canvas-sidebars/) for the bar and Pages in widgets to add form into bar (https://wordpress.org/plugins/pages-in-widgets/) -> possible to add forms in a sidebar and open via button click) . Only drawbacks i found so far: 
 a) links are somehow not the color of the theme
 b) width is not as you would suspect - so you have to try a lot
 c) the bar closes when you hit the submit-button - so you don't see the message that the form was submitted. If you reopen the bar you can see it.
Just wanted to leave this suggestions - as all those plugins are open source.
6. another suggestion: i looked into enquiry forms for woocommerce. Basically they are all "normal" forms you can already create with this plugin. Only thing is: they somehow automatically fill in the Subject. And subject is the product page you send the enquiry from (opened via button on the page). They are either popups (after hitting the button), showing under the button after you click enquiry or placed in an extra tab. I couldn't find anything regarding woocommerce in here. But would be a great addition.
